agent/consul/server_serf.go

```diff
@@ -177,6 +179,25 @@ func (s *Server) setupSerf(conf *serf.Config, ch chan serf.Event, path string, w
 	return serf.Create(conf)
 }
 
+func (s *Server) ReconnectTimeout(m *serf.Member, timeout time.Duration) time.Duration {
```

Comment:
Is there any difference between the Server and Client versions of `ReconnectTimeout` ? I couldn't find any.

The only thing these use from the method receiver is the logger. If we make this a function that accepts a logger, I think we can remove the duplicate copy.

If `ReconnectTimeoutOverride` accepted a function we could have passed a closure directly. Since it expects a type I guess we still need these methods, but they could both be a single line that calls into the shared function.

Reply:
They are the same I will create a common type for them.

Reply:
Nice!, I guess this method can be removed now as well

Reply:
Done.